cluster/ec2_banzaicloud.go

```diff
@@ -0,0 +1,504 @@
+	for _, role := range roles {
+		result = append(result, banzaicloudDB.Role(role))
+	}
+	return
+}
+
+func convertHosts(hosts banzaicloud.Hosts) (result banzaicloudDB.Hosts) {
+	for _, host := range hosts {
+		result = append(result, banzaicloudDB.Host{
+			Name:             host.Name,
+			PrivateIP:        host.PrivateIP,
+			NetworkInterface: host.NetworkInterface,
+			Roles:            convertRoles(host.Roles),
+			Labels:           convertLabels(host.Labels),
+			Taints:           convertTaints(host.Taints),
+		})
+	}
+
+	return
+}
+
+func convertNodePoolProvider(provider banzaicloud.NodePoolProvider) (result banzaicloudDB.NodePoolProvider) {
+	return banzaicloudDB.NodePoolProvider(provider)
+}
+
+func convertLabels(labels banzaicloud.Labels) banzaicloudDB.Labels {
+	res := make(banzaicloudDB.Labels, len(labels))
+	for k, v := range labels {
+		res[k] = v
+	}
+	return res
+}
+
+func convertTaints(taints banzaicloud.Taints) (result banzaicloudDB.Taints) {
+	for _, taint := range taints {
+		result = append(result, banzaicloudDB.Taint(taint))
+	}
+	return
+}
+
+func createEC2BanzaiCloudNetworkFromRequest(network banzaicloud.Network, userId uint) (banzaicloudDB.Network, error) {
+	n := banzaicloudDB.Network{
+		ServiceCIDR:      network.ServiceCIDR,
+		PodCIDR:          network.PodCIDR,
+		Provider:         convertNetworkProvider(network.Provider),
+		APIServerAddress: network.APIServerAddress,
+	}
+	n.CreatedBy = userId
+	return n, nil
+}
+
+func convertNetworkProvider(provider banzaicloud.NetworkProvider) (result banzaicloudDB.NetworkProvider) {
+	return banzaicloudDB.NetworkProvider(provider)
+}
+
+func createEC2BanzaiCloudKubernetesFromRequest(kubernetes banzaicloud.Kubernetes, userId uint) (banzaicloudDB.Kubernetes, error) {
+	k := banzaicloudDB.Kubernetes{
+		Version: kubernetes.Version,
+		RBAC:    banzaicloudDB.RBAC{Enabled: kubernetes.RBAC.Enabled},
+	}
+	k.CreatedBy = userId
+	return k, nil
+}
+
+func createEC2BanzaiCloudKubeADMFromRequest(kubernetes banzaicloud.KubeADM, userId uint) (banzaicloudDB.KubeADM, error) {
+	a := banzaicloudDB.KubeADM{
+		ExtraArgs: convertExtraArgs(kubernetes.ExtraArgs),
+	}
+	a.CreatedBy = userId
+	return a, nil
+}
+
+func convertExtraArgs(extraArgs banzaicloud.ExtraArgs) banzaicloudDB.ExtraArgs {
+	res := make(banzaicloudDB.ExtraArgs, len(extraArgs))
+	for k, v := range extraArgs {
+		res[k] = banzaicloudDB.ExtraArg(v)
+	}
+	return res
+}
+
+func createEC2BanzaiCloudCRIFromRequest(cri banzaicloud.CRI, userId uint) (banzaicloudDB.CRI, error) {
+	c := banzaicloudDB.CRI{
+		Runtime:       banzaicloudDB.Runtime(cri.Runtime),
+		RuntimeConfig: cri.RuntimeConfig,
+	}
+	c.CreatedBy = userId
+	return c, nil
+}
+
+func getMasterInstanceTypeAndImageFromNodePools(nodepools banzaicloudDB.NodePools) (masterInstanceType string, masterImage string, err error) {
+	for _, nodepool := range nodepools {
+		for _, role := range nodepool.Roles {
+			if banzaicloudDB.RoleMaster == role {
```

Comment:
Inside this if there is a dead code, banzaicloudDB.RoleMaster is a constant and the value is "master"